### PR TITLE
[SPARK-29323][WEBUI] Add tooltip for The Executors Tab's column names in the Spark history server Page

### PR DESCRIPTION
**Pull request by liucht-inspur**
### What changes were proposed in this pull request?

This PR is Adding tooltip for The Executors Tab's column names include RDD Blocks, Disk Used,Cores, Activity Tasks, Failed Tasks , Complete Tasks, Total Tasks in the history server Page.
https://issues.apache.org/jira/browse/SPARK-29323

image

I have modify the following code in executorspage-template.html
Before:
<th>RDD Blocks</th>
<th>Disk Used</th>
<th>Cores</th>
<th>Active Tasks</th>
<th>Failed Tasks</th>
<th>Complete Tasks</th>
<th>Total Tasks</th>

image

After:

<th><span data-toggle="tooltip" data-placement="top" title="RDD Blocks">RDD Blocks</span></th>
<th><span data-toggle="tooltip" data-placement="top" title="Disk Used">Disk Used</span></th>
<th><span data-toggle="tooltip" data-placement="top" title="Cores">Cores</span></th>
<th><span data-toggle="tooltip" data-placement="top" title="Active Tasks">Active Tasks</span></th>
<th><span data-toggle="tooltip" data-placement="top" title="Failed Tasks">Failed Tasks</span></th>
<th><span data-toggle="tooltip" data-placement="top" title="Complete Tasks">Complete Tasks</span></th>
<th><span data-toggle="tooltip" data-placement="top" title="Total Tasks">Total Tasks</span></th>
   

image
    

### Why are the changes needed?

the spark Executors of history Tab page, the Summary part shows the line in the list of title, but format is irregular.
Some column names have tooltip, such as Storage Memory, Task Time(GC Time), Input, Shuffle Read,
Shuffle Write and Blacklisted, but there are still some list names that have not tooltip. They are RDD Blocks, Disk Used,Cores, Activity Tasks, Failed Tasks , Complete Tasks and Total Tasks. oddly, Executors section below,All the column names Contains the column names above have tooltip .
It's important for open source projects to have consistent style and user-friendly UI, and I'm working on keeping it consistent And more user-friendly.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Manual tests for Chrome, Firefox and Safari

Authored-by: liucht-inspur <liucht@inspur.com>
Signed-off-by: liucht-inspur <liucht@inspur.com>

### CODE REVIEW

**Review by liucht-inspur — COMMENTED** [review]
Please help to review this change,thanks.

**Comment by srowen** [comment]
Where does the tooltip text come from? I'm missing that part.

**Comment by liucht-inspur** [comment]
@srowen They come from the same column names in the Executors like that under the red line: 
image

**Comment by liucht-inspur** [comment]
Summary code as

image


PK Executors code as

image
image

**Comment by liucht-inspur** [comment]
> Where does the tooltip text come from? I'm missing that part.

Like the Executors' column names

**Comment by srowen** [comment]
I mean, what text would the tooltip now show for the additional columns?

**Comment by liucht-inspur** [comment]
> I mean, what text would the tooltip now show for the additional columns?

I copy the code from Executors lable's column tooltip ，Before I revised  it, it is like this

Before modification
 Executors code as

image
image

**Comment by srowen** [comment]
I don't think that answers the question. What is the new tooltip that appears on what columns?

**Comment by liucht-inspur** [comment]
> Can one of the admins verify this patch?



> I don't think that answers the question. What is the new tooltip that appears on what columns?

Hello @srowen , I had screenshots and added comments, hope they are valuable to you,thanks!

image

**Comment by liucht-inspur** [comment]
@srowen My change is that 
the column **RDD Blocks** add tooltip text 'RDD Blocks',
the column **Disk Used** add tooltip text 'Disk Used',
the column **Cores** add tooltip text 'Cores',
the column **Active Tasks** add tooltip text 'Active Tasks',
the column **Failed Tasks** add tooltip text 'Failed Tasks',
the column **Complete Tasks** add tooltip text 'Complete Tasks',
the column **Total Tasks** add tooltip text 'Total Tasks',

**Comment by srowen** [comment]
What is the point of the tooltip then? it just repeats the column name.

**Comment by liucht-inspur** [comment]
> What is the point of the tooltip then? it just repeats the column name.

@srowen  It's important for open source projects to have consistent style and user-friendly UI, and I'm working on keeping it consistent And more user-friendly.

**Comment by srowen** [comment]
Hm, no I don't think that adds anything to user-friendliness. It's just a redundant popup. If you made the tooltips meaningful, maybe so.

**Comment by liucht-inspur** [comment]
So, can you give some suggestions about the tootip text?

**Comment by liucht-inspur** [comment]
> Hm, no I don't think that adds anything to user-friendliness. It's just a redundant popup. If you made the tooltips meaningful, maybe so.

@srowen I think it's necessary to be consistent in any case.

**Comment by srowen** [comment]
I don't think there's much meaningful to say in a tooltip here, so no I just wouldn't make this change.

**Comment by liucht-inspur** [comment]
@srowen For the same problem, I suggest adding more information in the columns of tooltip Active Tasks and Failed Tasks, because these two columns have different background colors, such as **blue**, **red**, and **white**, and they have **different transparency**, which is really confusing, what do you think?
See the columns of different background colors shown in the figure below
image

**Comment by srowen** [comment]
Yes, explaining the colors would be a reasonable role for a tooltip.

**Comment by liucht-inspur** [comment]
@srowen ok,thank you ! I have finished this change,

two changes:
1、in Summary 
before:
`<th>Active` `Tasks</th>`
`<th>Failed` `Tasks</th>`

after:
`<th><span` data-toggle="tooltip" data-placement="top" title="Active Tasks show in shaded blue. % of color opacity range from 0 to `maxTasks.">Active` `Tasks</span></th>`
`<th><span` data-toggle="tooltip" data-placement="top" title="Failed Tasks show in shaded red. Color opacity reach max at 10% of Total Tasks.">Failed `Tasks</span></th>`

added tooltip for Active Tasks and Failed Tasks ,the titles are 
"Active Tasks show in shaded blue. % of color opacity range from 0 to maxTasks." and "Failed Tasks show in shaded red. Color opacity reach max at 10% of Total Tasks."

2、in Executors
before:
`<th><span` data-toggle="tooltip" data-placement="top" title="Active Tasks">Active `Tasks</span></th>`
`<th><span` data-toggle="tooltip" data-placement="top" title="Failed Tasks">Failed `Tasks</span></th>`

after:
`<th><span` data-toggle="tooltip" data-placement="top" title="Active Tasks show in shaded blue. % of color opacity range from 0 to maxTasks.">Active `Tasks</span></th>`
`<th><span` data-toggle="tooltip" data-placement="top" title="Failed Tasks show in shaded red. Color opacity reach max at 10% of Total Tasks.">Failed `Tasks</span></th>`

added more information for tooltip

Please help to review this change,thanks.

File: core/src/main/resources/org/apache/spark/ui/static/executorspage-template.html [hunk 1]
```diff
@@ -37,8 +37,8 @@ <h4 class="title-table">Summary</h4>
         </th>
         <th>Disk Used</th>
         <th>Cores</th>
-        <th>Active Tasks</th>
-        <th>Failed Tasks</th>
+        <th><span data-toggle="tooltip" data-placement="top" title="Active Tasks show in shaded blue. % of color opacity range from 0 to maxTasks.">Active Tasks</span></th>
+        <th><span data-toggle="tooltip" data-placement="top" title="Failed Tasks show in shaded red. Color opacity reach max at 10% of Total Tasks.">Failed Tasks</span></th>
```
Comment (srowen) [inline core/src/main/resources/org/apache/spark/ui/static/executorspage-template.html]: Likewise, maybe:
"Number of tasks that have failed on this executor. Darker shading highlights executors with a high proportion of failed tasks."

**Review by srowen — CHANGES_REQUESTED** [review]

File: core/src/main/resources/org/apache/spark/ui/static/executorspage-template.html (hunk 1 above)
Comment (liucht-inspur) [reply core/src/main/resources/org/apache/spark/ui/static/executorspage-template.html]: Yes, they are more suitable，thank you！

**Comment by liucht-inspur** [comment]
@srowen Thanks again, I have finished the change.

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
We'll have to wait for Jenkins to come back, to evaluate this with tests, but I'm sure it will be OK.

**Comment by liucht-inspur** [comment]
cc @wangyum

**Comment by wangyum** [comment]
Sorry @liucht-inspur This has nothing to do with authorization:
http://apache-spark-developers-list.1001551.n3.nabble.com/build-system-IMPORTANT-northern-california-fire-danger-potential-power-outage-s-td28059.html

**Comment by liucht-inspur** [comment]
Ok, I see. We will wait quietly. Thank you all!

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
retest this please

**Comment by srowen** [comment]
Merged to master

**Comment by AbhishekNew** [comment]
@liucht-inspur Did u handle the tooltip for the Live UI Page for Spark.

**Comment by liucht-inspur** [comment]
> @liucht-inspur Did u handle the tooltip for the Live UI Page for Spark.

Yeah, exactly

**Pull request closed without merging**